aep/general/0146/aep.md.j2

```diff
@@ -1,5 +1,97 @@
 # Generic fields
 
-**Note:** This AEP has not yet been adopted. See
-[this GitHub issue](https://github.com/aep-dev/aep.dev/issues/27) for more
-information.
+Most fields in any API, whether in a request, a resource, or a custom response,
+have a specific type or schema. This schema is part of the contract that
+developers write their code against.
+
+However, occasionally it is appropriate to have a generic or polymorphic field
+of some kind that can conform to multiple schemata, or even be entirely
+free-form.
+
+## Guidance
+
+While generic fields are generally rare, a API **may** introduce generic field
+where necessary. There are several approaches to this depending on how generic
+the field needs to be; in general, APIs **should** attempt to introduce the
+"least generic" approach that is able to satisfy the use case.
+
+For example, an API **should not** use a completely generic field (such as
+`google.protobuf.Struct` in protobuf APIs) when the value of the field must
+correspond to one of a known number of schemas. Instead, the API **should** use
+a `oneof` to represent the known schemas.
```

Comment:
OneOf does exist in jsonschema: https://json-schema.org/draft/2020-12/json-schema-core#section-10.2.1.3

Reply:
From live discussion with @toumorokoshi / @earth2marsh / @makahmad:

* We need a unified version of the oneof concept.
* @toumorokoshi and I agree that the protobuf approach is preferable because clients can easily determine which case is being used.
* @earth2marsh suggests that at some point OAS Moonwalk may support non-JSON Schema modeling/schema languages.

Conclusion:  leave this as-is and file an issue to create an AEP on oneofs defining what they mean and how to implement them in both protobuf and OAS.  Link it here.